### PR TITLE
Factorize language selection into a single `LocaleChoiceType`

### PR DESCRIPTION
**Pull request by FabienPapet**
Signed-off-by: Fabien Papet <fabien.papet@gmail.com>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Introduce a new `LocaleChoiceType` to ease language selection. Factorized some code to make it more maintainable. 
| Type?             | new feature
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | All screens with language selection should behave as they used to.
| Fixed ticket?     |

### CODE REVIEW

**Review by matks — APPROVED** [review]
Good refactoring to decrease number of loc 👍

**Review by boherm — APPROVED** [review]
Nice idea @FabienPapet!

**Review by 0x346e3730 — APPROVED** [review]

**Review by HanaRebaiQA — APPROVED** [review]
Hello @FabienPapet 

I have checked your PR. 

- Check the automatic tests : OK
- Check with different languages : OK
- Check on different pages which have the language selection (Localization, Email theme, Translations etc) : OK
- Check with multistore : OK

So, it's QA :heavy_check_mark: 

Thank You!

File: src/PrestaShopBundle/Form/Admin/Type/LocaleChoiceType.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    protected function getLocaleChoices(): array
+    {
+        $locales = [];
+
+        foreach ($this->legacyContext->getLanguages() as $locale) {
+            $locales[$locale['name']] = $locale['iso_code'];
+        }
+
+        return $locales;
+    }
```
Comment (PululuK) [inline src/PrestaShopBundle/Form/Admin/Type/LocaleChoiceType.php:51]: Hi @FabienPapet , I suggest to move to a Choice provider 
See https://github.com/PrestaShop/PrestaShop/tree/develop/src/Adapter/Form/ChoiceProvider

Comment (FabienPapet) [reply src/PrestaShopBundle/Form/Admin/Type/LocaleChoiceType.php:51]: I'll do it in a dedicated PR, as this one is already QA :)

**Pull request merged**